packages/components/src/utils/event-guard.ts

```diff
@@ -1,48 +1,96 @@
-export function eventGuard(
-  host: HTMLElement,
-  event: CustomEvent,
-  options: { targetLocalName: string; delegatorSelector?: string },
-  callback: () => void
-): void {
-  const target = event.target as HTMLElement | null;
+/**
+ * Decorator that guards event handlers to only execute when the event meets certain criteria.
+ * @param options - Configuration options for the event guard
+ * @param options.targetLocalName - The local name of the element that should be the target of the event
+ * @param options.delegatorSelector - Optional CSS selector to check if the event was delegated from a specific ancestor
+ */
+export function EventGuard(options: { targetLocalName: string; delegatorSelector?: string }) {
+  return function (
+    target: unknown,
+    propertyKey: string,
+    descriptor?: PropertyDescriptor
+  ) {
+    if (descriptor) {
+      const originalMethod = descriptor.value;
+      
+      descriptor.value = function (event: CustomEvent) {
+        if (!event || !event.target) return;
+
+        const eventTarget = event.target as HTMLElement;
```

Comment:
You don’t know if the decorator will be correctly applied to a method that takes a custom event as a parameter, nor can you be sure if the event’s target will be an HTMLElement. To make your method foolproof, you need to test the types:
```typescript
      descriptor.value = function (event: unknown) {
        if (!(event instanceof CustomEvent && event.target instanceof HTMLElement)) return;
```

Also: will the decorator be used only for custom events, or can it also be applied to standard events?